### PR TITLE
[16.0] [MIG] partner_statement: Migration to 16.0

### PR DESCRIPTION
**Pull request by dannyadair**
This is basically https://github.com/OCA/account-financial-reporting/pull/1028 with the latest v15 merges

### CODE REVIEW

**Review by MiquelRForgeFlow — APPROVED** [review]

**Review by gdgellatly — APPROVED** [review]
Functional Review on live data

File: partner_statement/report/activity_statement_xlsx.py [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+
+    def _write_currency_lines(self, row_pos, sheet, partner, currency, data):
+        partner_data = data.get("data", {}).get(partner.id, {})
+        currency_data = partner_data.get("currencies", {}).get(currency.id)
+        account_type = data.get("account_type", False)
+        row_pos += 2
+        statement_header = _(
+            "%(payable)sStatement between %(start)s and %(end)s in %(currency)s"
+        ) % {
+            "payable": account_type == "payable" and _("Supplier ") or "",
```
Comment (MiquelRForgeFlow) [inline partner_statement/report/activity_statement_xlsx.py]:
```suggestion
            "payable": account_type == "liability_payable" and _("Supplier ") or "",
```

File: partner_statement/report/outstanting_statement_xlsx.py [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+            report_name = report_name + suffix
+        return report_name
+
+    def _write_currency_lines(self, row_pos, sheet, partner, currency, data):
+        partner_data = data.get("data", {}).get(partner.id, {})
+        currency_data = partner_data.get("currencies", {}).get(currency.id)
+        account_type = data.get("account_type", False)
+        row_pos += 2
+        statement_header = _("%(payable)sStatement up to %(end)s in %(currency)s") % {
+            "payable": account_type == "payable" and _("Supplier ") or "",
```
Comment (MiquelRForgeFlow) [inline partner_statement/report/outstanting_statement_xlsx.py]:
```suggestion
            "payable": account_type == "liability_payable" and _("Supplier ") or "",
```

File: partner_statement/views/activity_statement.xml [hunk 3]
```diff
@@ -0,0 +1,200 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2018 ForgeFlow, S.L. (https://www.forgeflow.com)
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+<odoo>
+    <template id="activity_balance">
+        <p>
+            <span t-if="is_detailed">Detailed </span>
+            <span
+                t-if="account_type == 'payable'"
```
Comment (MiquelRForgeFlow) [inline partner_statement/views/activity_statement.xml]:
```suggestion
                t-if="account_type == 'liability_payable'"
```

File: partner_statement/views/outstanding_statement.xml [hunk 4]
```diff
@@ -0,0 +1,190 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2018 ForgeFlow, S.L. (https://www.forgeflow.com)
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+<odoo>
+    <template id="outstanding_balance">
+        <p>
+            <span t-esc="'' if  account_type == 'receivable' else 'Supplier '" /><span
```
Comment (MiquelRForgeFlow) [inline partner_statement/views/outstanding_statement.xml]:
```suggestion
            <span t-esc="'' if account_type == 'asset_receivable' else 'Supplier '" /><span
```

File: partner_statement/report/detailed_activity_statement_xlsx.py [hunk 5]
```diff
@@ -0,0 +1,570 @@
...
+
+    def _write_currency_lines(self, row_pos, sheet, partner, currency, data):
+        partner_data = data.get("data", {}).get(partner.id, {})
+        currency_data = partner_data.get("currencies", {}).get(currency.id)
+        account_type = data.get("account_type", False)
+        row_pos += 2
+        statement_header = _(
+            "Detailed %(payable)sStatement between %(start)s and %(end)s in %(currency)s"
+        ) % {
+            "payable": account_type == "payable" and _("Supplier ") or "",
```
Comment (MiquelRForgeFlow) [inline partner_statement/report/detailed_activity_statement_xlsx.py]:
```suggestion
            "payable": account_type == "liability_payable" and _("Supplier ") or "",
```

**Comment by dannyadair** [comment]
Thanks for that @MiquelRForgeFlow I've now made those adjustments

**Comment by josep-tecnativa** [comment]
Please, include this changes https://github.com/OCA/account-financial-reporting/pull/1077/commits/a38bb838ae8c63af41dd3066e963198d3a49ed76

**Comment by dannyadair** [comment]
Thanks @josep-tecnativa - I wasn't sure how to cherrypick from another PR into my fork, also account_tax_balance is in a different state (this PR is only about partner_statement).

I've applied the changes to partner_statement tests. It is a separate commit, please let me know if it should be squashed into the migration commit.

**Comment by josep-tecnativa** [comment]
Hi @dannyadair , I have just noticed and it was my bad to include account_tax_balance in that commit. I think that it is okay if you just include partner_statement changes.

File: partner_statement/__manifest__.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
+# Copyright 2022 ForgeFlow, S.L. (http://www.forgeflow.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+{
+    "name": "Partner Statement",
+    "version": "16.0.1.1.0",
```
Comment (cuongnmtm) [inline partner_statement/__manifest__.py]:
```suggestion
    "version": "16.0.1.0.0",
```

Comment (dannyadair) [reply partner_statement/__manifest__.py]: Thanks @cuongnmtm I've now made that adjustment and squashed the last three commits back into one migration commit

**Comment by robinkeunen** [comment]
Hi @fclementic2c I think this is ready to be merged

**Comment by MiquelRForgeFlow** [comment]
Hi, I would like to supersede this PR with https://github.com/OCA/account-financial-reporting/pull/1106. It's just last clean-up before merge (includes https://github.com/OCA/account-financial-reporting/pull/1077). Can you close this and review https://github.com/OCA/account-financial-reporting/pull/1106?

**Comment by AaronHForgeFlow** [comment]
superseded

**Pull request closed without merging**